src/servers/src/http/otlp.rs

```diff
@@ -93,20 +87,139 @@ pub async fn traces(
     handler
         .traces(request, query_ctx)
         .await
-        .map(|o| OtlpTracesResponse {
+        .map(|o| OtlpResponse {
             resp_body: ExportTraceServiceResponse {
                 partial_success: None,
             },
             write_cost: o.meta.cost,
         })
 }
 
-pub struct OtlpTracesResponse {
-    resp_body: ExportTraceServiceResponse,
+pub struct PipelineInfo {
+    pub pipeline_name: Option<String>,
+    pub pipeline_version: Option<String>,
+}
+
+fn pipeline_header_error(
+    header: &HeaderValue,
+) -> StdResult<String, (http::StatusCode, &'static str)> {
+    match header.to_str() {
+        Ok(s) => Ok(s.to_string()),
+        Err(_) => Err((
+            StatusCode::BAD_REQUEST,
+            "`X-Pipeline-Name` or `X-Pipeline-Version` header is not string type.",
```

Comment:
to_str doesn't mean the value is not string. http headers are always string. we can return the nested error message.

Again, don't reinvent error handling in this module.